examples/side-effects/README.md

```diff
@@ -0,0 +1,280 @@
+This example shows how the `side-effects` flag for library authors works.
+
+The example contains a large library, `big-module`. `big-module` contains multiple child modules: `a`, `b` and `c`. The exports from the child modules are re-exported in the entry module (`index.js`) of the library. A consumer uses **some** of the exports, importing them from the library via `import { a, b } from "big-module"`. According to the EcmaScript spec, all child modules _must_ be evaluated because they could contain side effects.
+
+The `"side-effects": false` flag in `big-module`'s `package.json` indicates that the package's modules have no side effects (on evaluation) and only expose exports. This allows tools like webpack to optimize re-exports. In the case `import { a, b } from "big-module-with-flag"` is rewritten to `import { a } from "big-module-with-flag/a"; import { b } from "big-module-with-flag/b"`.
+
+The example contains two variants of `big-module`. `big-module` has no side-effects flag and `big-module-with-flag` has the side-effects flag. The example client imports `a` and `b` from each of the variants.
+
+After being built by webpack, the output bundle contains `index.js` `a.js` `b.js` `c.js` from `big-module`, but only `a.js` and `b.js` from `big-module-with-flag`.
+
+Advantages:
+
+* Smaller bundles
+* Faster bootup
+
+# example.js
+
+``` javascript
+import { a as a1, b as b1 } from "big-module";
+import { a as a2, b as b2 } from "big-module-with-flag";
+
+console.log(
+	a1,
+	b1,
+	a2,
+	b2
+);
+```
+
+# node_modules/big-module/package.json
+
+``` javascript
+{
+  "name": "big-module"
+}
+```
+
+# node_modules/big-module-with-flag/package.json
+
+``` javascript
+{
+  "name": "big-module-with-flag",
+  "side-effects": false
```

Comment:
it seems odd to use "side-effects" kebab case inside of package.json which is typically camelCased (e.g. devDependencies)

Reply:
(not a big deal, just an observation. I love this change!)

Reply:
I think this is a good idea @sokra, we probably want to normalize. 

Unless we are wanting to call out this is a webpack specific item.